chapters/solutions.Rmd

```diff
@@ -1430,6 +1430,95 @@ def test_fails_for_non_list():
 
 If the call to `total` *doesn't* raise `AssertionError`, this test fails.
 
+### Exercise \@ref(testing-ex-error-handling) {-}
+
+1. `collate.py` should read as follows:
```

Comment:
Looking at the book in print, we're repeating all of our scripts with slight variations several times - in cases like this, can we just show the bits that have changed? Or will that confuse?

Reply:
Good point. I think in general we need to give the full version of the script in the question/exercise (i.e. we can't expect the reader to look back and figure out what the script should look like), and then in the solutions only provide information about the bits that have changed. I've now done that in this PR and I'll create an issue to make sure we are being consistent with exercises and solutions in other chapters.